### PR TITLE
Feature/is healthiest via api

### PR DESCRIPTION
**Pull request by CyberDem0n**
https://github.com/zalando/patroni/issues/21

### CODE REVIEW

**Comment by jberkus** [comment]
At a conference right now, so I can't test this ... what does the JSON it returns look like?  I want to make sure it's extensible; when I add the ability to have designated failover replicas, I'll want to return some extra attributes.  As long as we're returning a JSON value where I can add more keys, that's fine.

**Comment by CyberDem0n** [comment]
{
    "running": true,
    "role": "master",
    "xlog": {
        "location": 6140462176
    },
    "postmaster_start_time": "2015-09-07 14:34:11.284 CEST"
}

{
    "running": true,
    "role": "slave",
    "xlog": {
        "received_location": 6140462176,
        "paused": false,
        "replayed_location": 6140462176
    },
    "postmaster_start_time": "2015-09-09 12:15:06.138 CEST"
}

File: patroni/ha.py [hunk 1]
```diff
@@ -41,6 +43,54 @@ def demote(self):
...
+        reachable - `!False` if the node is not reachable or is not responding with correct JSON
+        in_recovery - `!True` if pg_is_in_recovery() == true
+        xlog_location - value of `replayed_location` or `location` from JSON, dependin on its role."""
+
+        try:
+            response = requests.get(member.api_url, timeout=2)
+            json = response.json()
+            logger.info('Got response from %s: %s', member.name, json)
+            in_recovery = json.get('role', 'slave') != 'master'
+            xlog_location = json.get(('replayed_location' if in_recovery else 'location'), 0)
```
Comment (CyberDem0n) [inline patroni/ha.py]: Found a bug actually...
location and replayed_location are not in root node, but in a 'xlog'.

**Comment by jberkus** [comment]
Looks awesome, thanks.  I won't be able to test until tommorrow, though.

**Comment by jberkus** [comment]
Minor nitpick: can we use "replica" or "standby" instead of "slave"?  The word "slave" is kind of political in English.

**Comment by CyberDem0n** [comment]
Replaced slave with standby.

**Pull request closed without merging**